### PR TITLE
[release 2.0.1] [fix] fix load_sharded_optimizer_state_dict error on multi node

### PR DESCRIPTION
**Pull request by eracah**
cherry-picked from 4e26ad786d51b61003d48c2

cc @wz337

### CODE REVIEW

**Comment by atalman** [comment]
cc @eracah please fix lint error on the cherry-pick

**Comment by eracah** [comment]
> cc @eracah please fix lint error on the cherry-pick

Hi @atalman, to fix the lint error, it seems I have to cherry pick another commit which then requires another and another. I'm not sure how feasible this will be. Maybe @wz337 or @kumpera have a better idea on how to cleanly merge this one.

File: torch/distributed/checkpoint/optimizer.py [hunk 1]
```diff
@@ -169,19 +168,9 @@ def create_local_plan(self) -> LoadPlan:
...
-                            original_shard.metadata.shard_offsets, offset
-                        )
-                    ),
-                    sizes=torch.Size(original_shard.metadata.shard_sizes),
-                )
-            ]
-
-            reqs = create_read_items_for_chunk_list(
-                fqn, cast(TensorStorageMetadata, md), local_chunks
+            shard_md = copy.deepcopy(original_shard.metadata)
```
Comment (wz337) [inline torch/distributed/checkpoint/optimizer.py]: @kumpera Do we want to make this change in the main branch first? 
https://github.com/pytorch/pytorch/blob/main/torch/distributed/checkpoint/optimizer.py#L170-L183

**Review by wz337 — APPROVED** [review]
LGTM

**Review by atalman — APPROVED** [review]

**Pull request merged**